nupic/algorithms/KNNClassifier.py

```diff
@@ -143,6 +143,201 @@ def __init__(self,
     self.maxStoredPatterns = maxStoredPatterns
     self.clear()
 
+
+  @classmethod
+  def read(cls, proto):
+    knn = object.__new__(cls)
+
+    knn.version = proto.version
+    knn.verbosity = proto.verbosity
+
+    knn.cellsPerCol = proto.cellsPerCol
+    knn.k = proto.k
+    knn.distanceNorm = proto.distanceNorm
+    knn.distanceMethod = proto.distanceMethod
+    knn.distThreshold = proto.distanceThreshold
+    knn.doBinarization = proto.doBinarization
+    knn.binarizationThreshold = proto.binarizationThreshold
+    knn.useSparseMemory = proto.useSparseMemory
+    knn.sparseThreshold = proto.sparseThreshold
+    knn.relativeThreshold = proto.relativeThreshold
+    knn.numWinners = proto.numWinners
+    if proto.numSVDSamples != -1:
+      knn.numSVDSamples = proto.numSVDSamples
+    else:
+      knn.numSVDSamples = None
+    if proto.numSVDDims == 0:
+      knn._adaptiveSVDDims = True
+      knn.numSVDDims = 'adaptive'
+    else:
+      knn._adaptiveSVDDims = False
+      if proto.numSVDDims == -1:
+        knn.numSVDDims = None
+      else:
+        knn.numSVDDims = proto.numSVDDims
+    if proto.fractionOfMax != -1.:
```

Comment:
Not sure. But we could mitigate this risk replacing this to if int(proto.fractionOfMax) != -1: